File: Excel/300-399/316/316 Challenge.py
import pandas as pd

path = "300-399/316/316 Even Fibonacci Numbers.xlsx"
test = pd.read_excel(path, usecols="A", nrows=20)


def even_fibonacci(n):
    result = []
    a, b = 0, 1
    while len(result) < n:
        if a % 2 == 0:
            result.append(a)
        a, b = b, a + b
    return result


result = pd.DataFrame(
    {"First 20 Even Fibonnaci Numbers": even_fibonacci(20)}
)

print(result.equals(test))
# True

File: Excel/300-399/316/316 Challenge.R
library(tidyverse)
library(readxl)

path <- "300-399/316/316 Even Fibonacci Numbers.xlsx"
test <- read_excel(path, range = "A1:A21")

fib_even <- function(n) {
  result <- numeric(n)
  a <- 0
  b <- 1
  count <- 0
  while (count < n) {
    if (a %% 2 == 0) {
      count <- count + 1
      result[count] <- a
    }
    temp <- a + b
    a <- b
    b <- temp
  }
  result
}

result <- tibble(`First 20 Even Fibonnaci Numbers` = fib_even(20))

all.equal(result, test)
# [1] TRUE
